dev/helpers/mssql/read_required_migrations.sql

```diff
@@ -0,0 +1,20 @@
+CREATE OR ALTER PROCEDURE ReadRequiredMigrations
+    @migrationsFile NVARCHAR(MAX)
+AS
+BEGIN
+    SET NOCOUNT ON;
+
+    CREATE TABLE #InputMigrations (Filename NVARCHAR(MAX))
+
+    DECLARE @bulkInsertSql NVARCHAR(4000) = 'BULK INSERT #InputMigrations FROM ''' + @migrationsFile + ''' WITH (FIELDTERMINATOR = '';'', ROWTERMINATOR = '';'')';
```

Comment:
this requires the sql server to have access to the file which may not always be possible. Might be safer to parse the text file into json app side and pass in the json:

```
CREATE OR ALTER PROCEDURE ReadRequiredMigrations
    @migrationsJson NVARCHAR(MAX)
AS
BEGIN
    SET NOCOUNT ON;

    -- Create a table to hold the JSON data
    CREATE TABLE #InputMigrations (Filename NVARCHAR(MAX));

    -- Insert JSON data into the temporary table
    INSERT INTO #InputMigrations (Filename)
    SELECT [value] FROM OPENJSON(@migrationsJson);

    -- The rest of your procedure remains the same
    SELECT IM.[Filename]
    FROM [#InputMigrations] IM
    LEFT JOIN [dbo].[migrations] M ON IM.[Filename] = M.[Filename]
    WHERE M.[Filename] IS NULL
    ORDER BY [Filename] ASC;

    DROP TABLE #InputMigrations;
END;
```